std/math.d

```diff
@@ -4590,7 +4590,7 @@ public:
     assert(ieeeFlags == f);
 }
 
-@system unittest
+version(D_HardFloat) @system unittest
```

Comment:
Only one of the commits is fine, we can merge it now without waiting on the other two.  I'm not sure about this change, but I haven't looked at the body, and I don't wish to discuss here, as the topic is about murmurhash.